File: frontend/src/api/stock.ts
import api from './client';

export interface Stock {
  id: string;
  item_id: string;
  location_id: string;
  quantity: number;
  reserved_quantity: number;
  batch_number: string;
  item_sku?: string;
  item_name?: string;
  location_code?: string;
}

export interface AdjustStockRequest {
  item_id: string;
  location_id: string;
  quantity: number;
  notes?: string;
}

export interface StockOpnameRequest {
  location_id: string;
  notes?: string;
  items: StockOpnameItemRequest[];
}

export interface StockOpnameItemRequest {
  item_id: string;
  system_quantity: number;
  actual_quantity: number;
}

export interface StockMovement {
  id: string;
  item_id: string;
  from_location_id?: string;
  to_location_id?: string;
  quantity: number;
  movement_type: string;
  reference_type: string;
  reference_id: string;
  notes: string;
  created_by: string;
}

export const stockApi = {
  list: (page = 1, limit = 10, itemId?: string, locationId?: string, search?: string) => {
    const params: any = { page, limit };
    if (itemId) params.item_id = itemId;
    if (locationId) params.location_id = locationId;
    if (search) params.search = search;
    return api.get('/stock', { params });
  },
  getByItem: (itemId: string) =>
    api.get<Stock[]>('/stock', { params: { item_id: itemId } }),
  adjust: (data: AdjustStockRequest) => api.post('/stock/adjust', data),
  opname: (data: StockOpnameRequest) => api.post('/stock/opname', data),
  listMovements: (page = 1, limit = 10) =>
    api.get('/stock/movements', { params: { page, limit } }),
};

File: backend/domain/entity/stock.go
package entity

import "time"

type Stock struct {
	ID               string     `json:"id"`
	ItemID           string     `json:"item_id"`
	LocationID       string     `json:"location_id"`
	Quantity         float64    `json:"quantity"`
	ReservedQuantity float64    `json:"reserved_quantity"`
	BatchNumber      string     `json:"batch_number"`
	ExpiryDate       *time.Time `json:"expiry_date,omitempty"`
	UpdatedAt        time.Time  `json:"updated_at"`
}

type StockMovement struct {
	ID             string    `json:"id"`
	ItemID         string    `json:"item_id"`
	FromLocationID *string   `json:"from_location_id,omitempty"`
	ToLocationID   *string   `json:"to_location_id,omitempty"`
	Quantity       float64   `json:"quantity"`
	MovementType   string    `json:"movement_type"`
	ReferenceType  string    `json:"reference_type"`
	ReferenceID    string    `json:"reference_id"`
	Notes          string    `json:"notes"`
	CreatedBy      string    `json:"created_by"`
	CreatedAt      time.Time `json:"created_at"`
}
